5-infrastructure/modules/base-env/variables.tf

```diff
@@ -0,0 +1,359 @@
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+variable "env" {
+  type        = string
+  description = "The environment to prepare (ex. development)."
+}
+
+variable "env_short" {
+  type        = string
+  description = "The environment to prepare (ex. dev)."
+}
+
+variable "environment_code" {
+  type        = string
+  description = "A short form of the folder level resources (environment) within the Google Cloud organization (ex. d)."
+}
+
+variable "business_unit" {
+  type        = string
+  description = "A short form of the business unit level projects within the Google Cloud organization (ex. bu1)."
+  default     = "bu1"
+}
+
+variable "parent_folder" {
+  type        = string
+  description = "The parent folder or org for environments."
+}
+
+variable "terraform_service_account" {
+  type        = string
+  description = "Service account email of the account to impersonate to run Terraform."
+}
+
+variable "project_prefix" {
+  type        = string
+  description = "Name prefix to use for projects created."
+  default     = "prj"
+}
+
+variable "folder_prefix" {
+  type        = string
+  description = "Name prefix to use for folders created."
+  default     = "fldr"
+}
+
+variable "location_primary" {
+  type        = string
+  description = "The primary region for deployment, if not set default locations for each resource are taken from variables file"
+  default     = "us-east1"
+}
+
+variable "location_secondary" {
+  type        = string
+  description = "The secondary region for deployment, if not set default locations for each resource are taken from variables file"
+  default     = "us-west1"
+}
+
+variable "log_storage_bucket_location" {
+  type        = string
+  description = "The region the storage bucket for logs is located."
+  default     = "US-WEST1"
+}
+
+variable "log_storage_bucket_project" {
+  type        = string
+  description = "The project that will contain the storage bucket for logs, if not set, OPS project is taken by default"
+  default     = ""
+}
+
+variable "kms_location_1" {
```

Comment:
rather than having all locations for each resource parameterizes, lets just parameterize 2 vars as location1 and location2 that will be used throughout.

Reply:
primary and secondary location variables exist, if not given, modules allow to set each resource location seperately

Reply:
@bharathkkb sql module shifted to for_each approch please review and I can continue with others ae4e0e02c2b6a33cd5d6a2897630adb7b7f9caaa

Reply:
> modules allow to set each resource location seperately

For this demo we don't need this ability. It is trivial to add this from a users perspective